benchmarks/test_correlation_MG3_SD.py

```diff
@@ -0,0 +1,158 @@
+import os
+import numpy as np
+import pyccl as ccl
+from scipy.interpolate import interp1d
+import pytest
+
+
+@pytest.fixture(scope='module', params=['fftlog', 'bessel'])
+def corr_method(request):
+    errfacs = {'fftlog': 0.21, 'bessel': 0.05}
+    return request.param, errfacs[request.param]
+
+
+@pytest.fixture(scope='module')
+def set_up(request):
+    dirdat = os.path.dirname(__file__) + '/data'
+    h0 = 0.67702026367187500
+    logA = 3.05  # log(10^10 A_s)
+    # scale dependent MG cosmology
+    cosmo = ccl.Cosmology(Omega_c=0.12/h0**2, Omega_b=0.0221/h0**2, Omega_k=0,
+                          h=h0, A_s=np.exp(logA)/10**10, n_s=0.96, Neff=3.046,
+                          m_nu=0.0, w0=-1, wa=0, T_CMB=2.7255,
+                          mu_0=0.1, sigma_0=0.1,
+                          c1_mg=1.1, c2_mg=1.1, lambda_mg=1,
+                          transfer_function='boltzmann_isitgr',
+                          matter_power_spectrum='linear')
+    cosmo.cosmo.gsl_params.INTEGRATION_LIMBER_EPSREL = 2.5E-5
+    cosmo.cosmo.gsl_params.INTEGRATION_EPSREL = 2.5E-5
+
+    # Ell-dependent correction factors
+    # Set up array of ells
+    fl = {}
+    lmax = 10000
+    nls = (lmax - 400)//20+141
+    ells = np.zeros(nls)
+    ells[:101] = np.arange(101)
+    ells[101:121] = ells[100] + (np.arange(20) + 1) * 5
+    ells[121:141] = ells[120] + (np.arange(20) + 1) * 10
+    ells[141:] = ells[140] + (np.arange(nls - 141) + 1) * 20
+    fl['lmax'] = lmax
+    fl['ells'] = ells
+
+    # Load dNdz's
+    z1, pz1 = np.loadtxt(dirdat + "/bin1_histo.txt", unpack=True)
+    z2, pz2 = np.loadtxt(dirdat + "/bin2_histo.txt", unpack=True)
+
+    # Set up the linear galaxy bias as used in generating benchmarks
+    bz1 = 1.45*np.ones_like(pz1)
+    bz2 = 1.55*np.ones_like(pz2)
+
+    # Initialize tracers
+    trc = {}
+    trc['g1'] = ccl.NumberCountsTracer(cosmo, False, (z1, pz1), (z1, bz1))
+    trc['g2'] = ccl.NumberCountsTracer(cosmo, False, (z2, pz2), (z2, bz2))
+    trc['l1'] = ccl.WeakLensingTracer(cosmo, (z1, pz1))
+    trc['l2'] = ccl.WeakLensingTracer(cosmo, (z2, pz2))
+
+    # Read benchmarks
+    bms = {}
+    bms['dd_11'] = np.loadtxt(dirdat+'/wtheta_linear_predictionSD.dat')[0:15]
```

Comment:
I don't think the scale-dependent benchmark files have been uploaded to the repo.

Also, how were the benchmarks created?

Reply:
Not sure what happened. I thought I did but apparently the data files stayed behind (due to the ignore). I just uploaded all of them with add -f so it worked.  The benchmarks where created using ISiTGR.

Reply:
Ok, thanks @damonge for the third batch of comments and I have addressed all them as well as the new suggestions. 
I also addressed @c-d-leonard comments and the last ones about flake8. 
Thanks to both of you for all the comments and suggestions. I hope we are ready to move forward.